DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs

```diff
@@ -1841,7 +1839,11 @@ public override void UpdateUsersOnline(Hashtable userList)
                 {
                     //Clear the user object
                     user = null;
-                }                              
+
+                    //Clear cache for user so that locked out & other status could be updated
+                    DataCache.ClearUserCache(portalId, username);
+                    DataCache.ClearCache(GetCacheKey(username));
```

Comment:
What do those two calls clear? Untrained eye could think that the first call should be enough to "clear user cache"